go/tools/builders/link.go

```diff
@@ -141,6 +141,16 @@ func link(args []string) error {
 	}
 	goargs = append(goargs, "-o", *outFile)
 
+	// substitute `builder cc` for the linker with a script. unfortunately
+	// we can't just set an environment variable to `builder cc` because
+	// in `go tool link` the `linkerFlagSupported` call sites used to determine
```

Comment:
Could you add some references to the source code here? That would make it easier to look for alternative solutions.

Reply:
I'm not quite sure if you want references to the code in the comment above or just here in github but here's the relevant line:

https://cs.opensource.google/go/go/+/master:src/cmd/link/internal/ld/lib.go;l=1732?q=hostlink

You can see the call to `linkerFlagSupported()` uses `argv[0]` as the `linker`. `argv` is set up here:

https://cs.opensource.google/go/go/+/master:src/cmd/link/internal/ld/lib.go;drc=f2d118fd5f7e872804a5825ce29797f81a28b0fa;l=479?q=hostlink

This uses the global `flagExtLd` which is a `quoted.Flag`, a type defined here:

https://cs.opensource.google/go/go/+/master:src/cmd/internal/quoted/quoted.go;drc=f2d118fd5f7e872804a5825ce29797f81a28b0fa;l=107

This type is a `[]string` which means that `flag.Var` will parse this value into a slice of strings, resulting in something like `[]string{"/path/to/builder", "cc"}`.

Reply:
Could you add the first two to the comment? Thanks for looking them up!

Reply:
np, done